File: app/src/main/java/com/uralsiberianworks/neuralpushkin/db/ContactDao.java
package com.uralsiberianworks.neuralpushkin.db;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ContactDao {

    @Query("SELECT * FROM contact")
    List<Contact> getAllContacts();

    @Query("SELECT * FROM contact WHERE contactID = :id")
    Contact getContact(String id);

    @Insert
    void insert(Contact contact);

    @Update
    void update(Contact contact);


    @Query("DELETE FROM contact WHERE contactID = :id")
    void del(String id);
}

File: app/src/main/java/com/uralsiberianworks/neuralpushkin/database/ContactDao.kt
package com.uralsiberianworks.neuralpushkin.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ContactDao {
    @get:Query("SELECT * FROM contact")
    val allContacts: MutableList<Contact?>?

    @Query("SELECT * FROM contact WHERE contactID = :id")
    fun getContact(id: String?): Contact?

    @Insert
    fun insert(contact: Contact?)

    @Update
    fun update(contact: Contact?)

    @Query("DELETE FROM contact WHERE contactID = :id")
    fun del(id: String?)
}